### PR TITLE
[OKTA-460124] "Protect Your API" PHP Quickstart 

### PR DESCRIPTION
**Pull request by danielmaharry-okta**
## Description:
- **What's changed?** 
- Added "Protect Your API" quickstart for PHP by Aaron Parecki. Copied from Google Docs

- **Is this PR related to a Monolith release?** No

### Resolves:
* [OKTA-460124](https://oktainc.atlassian.net/browse/OKTA-460124)

Preview URL : https://622f65e9374a950c2bd9539b--reverent-murdock-829d24.netlify.app/docs/guides/protect-your-api/php/main/

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/protect-your-api/main/php/reqautheverything.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      ;
+  } catch(\Exception $e) {
+    return false;
+  }
+
+  return true;
+}
+```
+
+This function calls another function that will need to be created which is responsible for fetching and caching the JWT keys from Okta, so create that function now:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/protect-your-api/main/php/reqautheverything.md:32]:
```suggestion
This function calls another function that you need to create that is responsible for fetching and caching the JWT keys from Okta. Create that function now:
```

File: packages/@okta/vuepress-site/docs/guides/protect-your-api/main/php/reqauthspecific.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+Delete or comment out the code block you added above that calls `hasValidAccessToken()` and instead move that into the particular API routes you want to protect. For example, you can protect only the `whoami` route and output the claims of the access token. Replace your `whoami` function with the below.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/protect-your-api/main/php/reqauthspecific.md:1]:
```suggestion
Delete or comment out the code block you added above that calls `hasValidAccessToken()` and instead move that into the particular API routes that you want to protect. For example, you can protect only the `whoami` route and output the claims of the access token. Replace your `whoami` function with the following:
```

**Review by susanharper-okta — APPROVED** [review]
couple nits. Approving to not hold you up. FYI - in the Set up Okta section - between steps 4 and 5 - that code snippet and note should be indented to aline with the step numbering.

**Review by chrisdmills-okta — APPROVED** [review]
Hi @danielmaharry-okta ! This is looking jolly good; just a couple of bits that need updating before you put this live:

1. Make PHP appear in the "Protect your API" list on the homepage — https://developer.okta.com/. This is done by uncommenting the relevant frontmatter info in https://github.com/okta/okta-developer-docs/blob/master/packages/%40okta/vuepress-site/index.md
2. Remove "coming soon" from the relevant PHP card on https://developer.okta.com/docs/guides/quickstart/main/#protect-your-api

File: packages/@okta/vuepress-site/docs/guides/protect-your-api/main/php/createroute.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    break;
+
+  case '/api/hello':
+    hello();
+    break;
+
+}
+```
+
+The two functions `whoami()` and `hello()` are where you would define the actual logic of your API methods. For now, you can define these functions to just output a string of text.
```
[the remark this answers is not in the document]
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/protect-your-api/main/php/createroute.md:24]: fixed

File: packages/@okta/vuepress-site/docs/guides/protect-your-api/main/php/independ.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+This guide uses several libraries which will need to be installed before continuing. The easiest way to install these libraries is to use [Composer](https://getcomposer.org), so install that on your system before continuing to install the dependencies.
```
[the remark this answers is not in the document]
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/protect-your-api/main/php/independ.md:1]: fixed

File: packages/@okta/vuepress-site/docs/guides/protect-your-api/main/php/independ.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+```
+
+Next, install the Web Token JWT library:
+
+```bash
+composer require web-token/jwt-easy ^2
+composer require web-token/jwt-signature-algorithm-rsa ^2
+```
+
+To fetch Okta's JWT signing keys, this quickstart will use the Guzzle HTTP client:
```
[the remark this answers is not in the document]
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/protect-your-api/main/php/independ.md:16]: fixed

File: packages/@okta/vuepress-site/docs/guides/protect-your-api/main/php/independ.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+```bash
+composer require guzzlehttp/guzzle ^7
+```
+
+As explained later, it's a good idea to cache the public key to speed up access token validation. You should probably use whatever caching mechanism is available in the particular framework you're using for your API, but to keep things simple, this quickstart will use a file-based cache.
```
[the remark this answers is not in the document]
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/protect-your-api/main/php/independ.md:22]: fixed

File: packages/@okta/vuepress-site/docs/guides/protect-your-api/main/php/reqautheverything.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
+First, create a function that will check whether a request has a valid access token. This function will be responsible for parsing the incoming `Authorization` header and validating the JWT.
```
[the remark this answers is not in the document]
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/protect-your-api/main/php/reqautheverything.md:1]: fixed

File: packages/@okta/vuepress-site/docs/guides/protect-your-api/main/php/reqautheverything.md (hunk 1 above)
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/protect-your-api/main/php/reqautheverything.md:32]: fixed

File: packages/@okta/vuepress-site/docs/guides/protect-your-api/main/php/reqauthspecific.md (hunk 2 above)
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/protect-your-api/main/php/reqauthspecific.md:1]: fixed

**Comment by danielmaharry-okta** [comment]
@chrisdmills-okta Changes all made. Merge when needed.

**Comment by chrisdmills-okta** [comment]
@danielmaharry-okta boom!

**Pull request merged**